custom/icds/views/hosted_ccz.py

```diff
@@ -87,6 +88,37 @@ def post(self, request, *args, **kwargs):
         return self.get(request, *args, **kwargs)
 
 
+@require_GET
+@toggles.MANAGE_CCZ_HOSTING.required_decorator()
+def ccz_hostings_json(request, domain):
+    limit = int(request.GET.get('limit', 10))
+    page = int(request.GET.get('page', 1))
+
+    if request.GET.get('link_id'):
+        hosted_cczs = HostedCCZ.objects.filter(link_id=request.GET.get('link_id'))
+    else:
+        hosted_cczs = HostedCCZ.objects.filter(link__domain=domain)
+    if request.GET.get('app_id'):
+        hosted_cczs = hosted_cczs.filter(app_id=request.GET.get('app_id'))
+    version = request.GET.get('version')
+    if version:
+        hosted_cczs = hosted_cczs.filter(version=request.GET.get('version'))
+    if request.GET.get('profile_id'):
+        hosted_cczs = hosted_cczs.filter(profile_id=request.GET.get('profile_id'))
+    if request.GET.get('status'):
+        hosted_cczs = hosted_cczs.filter(status=request.GET.get('status'))
+
+    total = len(hosted_cczs)
```

Comment:
does len already calls `count` internally?
if not, we should do that to avoid fetching all records and then just fetching the ones we need?

Reply:
good point, 56c96f5f3da